addons/mail/static/src/models/welcome_view/welcome_view.js

```diff
@@ -0,0 +1,127 @@
+import { registerNewModel } from '@mail/model/model_core';
+
+function factory(dependencies) {
+
+    class WelcomeView extends dependencies['mail.model'] {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @static
+         * @param {string} guestName 
+         * @returns {string}
+         */
+        static sanitizeGuestName(guestName = "") {
+            return guestName.trim();
+        }
+
+        /**
+         * @static
+         * @param {string} guestName 
+         */
+        static async updateGuestNameServerSide(guestName) {
+            await this.env.services.rpc({
+                route: '/mail/guest/update_name',
+                params: {
+                    guest_name: this.messaging.models['mail.welcome_view'].sanitizeGuestName(guestName),
+                },
+            });
+        }
+
+        /**
+         * Redirects users to the related channel. If the current user is a
+         * guest, first updates their username with the value of guestNameInput.
+         */
+        async updateGuestNameIfAnyThenJoinChannel() {
+            if (this.messaging.currentGuest) {
+                if (!this.isGuestNameValid) {
+                    return;
+                }
+                await this.messaging.models['mail.welcome_view'].updateGuestNameServerSide(this.guestName);
+            }
+            window.location = `/discuss/channel/${this.channelId}`;
+        }
+
+        /**
+         * @param {MouseEvent} ev 
+         */
+        onClickJoinButton(ev) {
+            this.updateGuestNameIfAnyThenJoinChannel();
+        }
+
+        /**
+         * Handles OWL update on this WelcomeView component.
+         */
+        onComponentUpdate() {
+            this._handleFocus();
+        }
+
+        /**
+         * @param {KeyboardEvent} ev 
+         */
+        onInputGuestNameInput(ev) {
+            this._updateGuestNameWithInputValue();
+        }
+
+        /**
+         * @param {KeyboardEvent} ev 
+         */
+        onKeydownGuestNameInput(ev) {
+            if (ev.key === 'Enter') {
+                this.updateGuestNameIfAnyThenJoinChannel();
+            }
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         */
+        _handleFocus() {
+            if (this.isDoFocusGuestNameInput) {
+                this.update({ isDoFocusGuestNameInput: false });
+                if (this.guestNameInputRef.el) {
+                    this.guestNameInputRef.el.focus();
+                }
+            }
+        }
+
+        /**
+         * @private
+         */
+        _updateGuestNameWithInputValue() {
+            if (this.guestNameInputRef.el) {
```

Comment:
It is actually possible that we are on the input event and the element is not actually written correctly on the record?